### PR TITLE
fix: finishing touches to sync history

### PR DESCRIPTION
**Pull request by kiram15**
# For all changes

[Screen Recording 2023-05-05 at 4.46.43 PM.zip](https://github.com/openedx/frontend-app-admin-portal/files/11422308/Screen.Recording.2023-05-05.at.4.46.43.PM.zip)

- [x] Ensure adequate tests are in place (or reviewed existing tests cover changes)

# Only if submitting a visual change

- [x] Ensure to attach screenshots
- [x] Ensure to have UX team confirm screenshots

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -303,47 +303,51 @@ const channelMapping = {
   [BLACKBOARD_TYPE]: {
     displayName: 'Blackboard',
     icon: BlackboardIcon,
-    post: LmsApiService.postNewBlackboardConfig,
```
Comment (kiram15) [inline src/utils.js:306]: nit: alphabetizing

File: src/components/settings/SettingsLMSTab/ErrorReporting/SyncHistory.jsx [hunk 2]
```diff
@@ -2,38 +2,58 @@ import React, { useEffect, useState } from 'react';
...
 const SyncHistory = () => {
+  // the simple redirect is used for going back to the lms page
   const vars = (window.location.pathname).split('lms/');
   const redirectPath = `${vars[0]}lms/`;
   const configInfo = vars[1].split('/');
   const configChannel = configInfo[0];
   const configId = configInfo[1];
 
+  // the redirect with params is used when editing an existing config
+  let fullLink = `${((window.location.href).split('lms/'))[0]}lms/?`;
```
Comment (marlonkeating) [inline src/components/settings/SettingsLMSTab/ErrorReporting/SyncHistory.jsx:31]: nit: More descriptive name would be nice in the case that other links get added down the line, maybe something like 'editConfigUrl'.

File: src/components/settings/SettingsLMSTab/ErrorReporting/SyncHistory.jsx [hunk 3]
```diff
@@ -2,38 +2,58 @@ import React, { useEffect, useState } from 'react';
...
   const [errorIsOpen, openError, closeError] = useToggle(false);
   const [showDeleteModal, setShowDeleteModal] = useState(false);
   const [toastMessage, setToastMessage] = useState(null);
   const [reloadPage, setReloadPage] = useState(false);
 
-  const getActiveStatus = status => (status === 'Active' ? `${status} •` : '');
+  const getSubheaders = () => {
+    const sub1 = (getStatus(config) === 'Active' ? `${getStatus(config)}` : null);
+    const sub2 = channelMapping[config.channelCode].displayName;
+    const sub3 = `Last modified on ${formatTimestamp({ timestamp: config.lastModifiedAt })}`;
```
Comment (marlonkeating) [inline src/components/settings/SettingsLMSTab/ErrorReporting/SyncHistory.jsx:48]: nit: Descriptive variable names would be nice here too

File: src/components/settings/SettingsLMSTab/index.jsx [hunk 4]
```diff
@@ -62,7 +61,21 @@ const SettingsLMSTab = ({
     setShowNoConfigCard(false);
     // Since the user is editing, hide the existing config cards
     setConfigsExist(false);
-  };
+    openLmsStepper();
+  }, [dispatch, openLmsStepper]);
```
Comment (marlonkeating) [inline src/components/settings/SettingsLMSTab/index.jsx:65]: We aren't using `openLmsStepper`in this callback anymore, right?
```suggestion
  }, [dispatch, configData]);
```

**Review by marlonkeating — CHANGES_REQUESTED** [review]

File: src/components/settings/SettingsLMSTab/index.jsx (hunk 4 above)
Comment (kiram15) [reply src/components/settings/SettingsLMSTab/index.jsx:65]: Open lms stepper is still being used, see line 64

File: src/components/settings/SettingsLMSTab/ErrorReporting/SyncHistory.jsx [hunk 5]
```diff
@@ -230,6 +230,9 @@ const SyncHistory = () => {
                   src={channelMapping[configChannel].icon}
                 />
                 {config.displayName}
+                {getStatus(config) !== 'Active' && (
+                  <Badge className="card-status-badge" variant="light">{getStatus(config)}</Badge>
```
Comment (marlonkeating) [inline src/components/settings/SettingsLMSTab/ErrorReporting/SyncHistory.jsx:234]: nit: I know this was an existing issue in this file, but `getStatus(config)` is used now 7 times.  Can we memo-ize this value once in this component?

File: src/components/settings/SettingsLMSTab/index.jsx (hunk 4 above)
Comment (marlonkeating) [reply src/components/settings/SettingsLMSTab/index.jsx:65]: Gotcha, not sure how I missed that...

**Review by marlonkeating — APPROVED** [review]

**Pull request merged**